livekit-plugins/livekit-plugins-fal/livekit/plugins/fal/sd_turbo_high_fps.py

```diff
@@ -0,0 +1,211 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import io
+import logging
+import os
+from dataclasses import dataclass
+from typing import Optional, Set
+
+import aiohttp
+import msgpack
+from livekit import rtc
+from PIL import Image
+
+FAL_URL = "wss://110602490-sd-turbo-real-time-high-fps-msgpack.gateway.alpha.fal.ai/ws"
+
+
+class SDTurboHighFPS:
+    """Fal SDXL Plugin
+
+    Requires FAL_KEY_ID and FAL_KEY_SECRET environment variables to be set.
+    """
+
+    def __init__(
+        self,
+        *,
+        key_id: Optional[str] = None,
+        key_secret: Optional[str] = None,
+    ):
+        self._key_id = key_id or os.environ.get("FAL_KEY_ID")
+        self._key_secret = key_secret or os.environ.get("FAL_KEY_SECRET")
+
+    def stream(self) -> "SDTurboHighFPSStream":
+        return SDTurboHighFPSStream(key_id=self._key_id, key_secret=self._key_secret)
+
+
+def _task_done_cb(task: asyncio.Task, set: Set[asyncio.Task]) -> None:
+    set.discard(task)
+    if task.cancelled():
+        logging.info("task cancelled: %s", task)
+        return
+
+    if task.exception():
+        logging.error("task exception: %s", task, exc_info=task.exception())
+        return
+
+
+EOS: None = None
+
+
+class SDTurboHighFPSStream:
+    @dataclass
+    class Input:
+        data: bytes
+        prompt: str
+        strength: float
+        width: int
+        height: int
+
+    def __init__(self, key_id: str, key_secret: str):
+        self._key_id = key_id
+        self._key_secret = key_secret
+        self._session = aiohttp.ClientSession()
+        self._ws: Optional[aiohttp.ClientWebSocketResponse] = None
+        self._input_queue = asyncio.Queue[self.Input]()
+        self._output_queue = asyncio.Queue[rtc.VideoFrame]()
+        self._in_flight_requests = 0
+        self._tasks = set()
+        self._closed = False
+        run_task = asyncio.create_task(self._run())
+        run_task.add_done_callback(lambda t: _task_done_cb(t, self._tasks))
+
+    def push_frame(self, frame: rtc.VideoFrame, prompt: str, strength: float) -> None:
+        if self._closed:
+            raise ValueError("cannot push to a closed stream")
+
+        argb_frame = rtc.ArgbFrame.create(
+            format=rtc.VideoFormatType.FORMAT_RGBA,
+            width=frame.buffer.width,
+            height=frame.buffer.height,
+        )
+        frame.buffer.to_argb(dst=argb_frame)
+        self._input_queue.put_nowait(
+            self.Input(
+                data=argb_frame.data,
+                prompt=prompt,
+                strength=strength,
+                width=frame.buffer.width,
+                height=frame.buffer.height,
+            )
+        )
+
+    async def aclose(self) -> None:
+        self._closed = True
```

Comment:
You should wait for the end of task, otherwise asyncio will report a leak when closing the program after that